src/main/java/teammates/ui/controller/AdminSessionsPageAction.java

```diff
@@ -101,35 +97,35 @@ private ActionResult validateParametersAndCreateShowPageResultIfInvalid() {
             String endMin = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_ENDMINUTE);
             String timeZone = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_TIMEZONE);
 
-            zone = ZoneId.of(timeZone);
-            start = TimeHelper.parseLocalDateTimeForSessionsForm(startDate, startHour, startMin);
-            end = TimeHelper.parseLocalDateTimeForSessionsForm(endDate, endHour, endMin);
-
-            if (start.isAfter(end)) {
-                isError = true;
-                statusToUser.add(new StatusMessage("The filter range is not valid."
-                        + " End time should be after start time.", StatusMessageColor.DANGER));
-                statusToAdmin = "Admin Sessions Page Load<br>"
-                        + "<span class=\"bold\"> Error: invalid filter range</span>";
-                prepareDefaultPageData(start, end);
-                return createShowPageResultWithDataInitialization();
-            }
+            this.rangeStart = TimeHelper.parseLocalDateTimeForSessionsForm(startDate, startHour, startMin);
+            this.rangeEnd = TimeHelper.parseLocalDateTimeForSessionsForm(endDate, endHour, endMin);
+            this.timeZone = ZoneId.of(timeZone);
 
-            this.rangeStart = start;
-            this.rangeEnd = end;
-            this.timeZone = zone;
-            return null;
+            if (!this.rangeStart.isAfter(this.rangeEnd)) {
+                return null;
+            }
         }
 
         isError = true;
-        statusToUser.add(new StatusMessage("Error: Missing Parameters", StatusMessageColor.DANGER));
-        statusToAdmin = "Admin Sessions Page Load<br>"
-                + "<span class=\"bold\"> Error: Missing Parameters</span>";
-        prepareDefaultPageData(start, end);
-        return createShowPageResultWithDataInitialization();
+        if (this.rangeStart.isAfter(this.rangeEnd)) {
```

Comment:
`this.rangeStart` and `this.rangeEnd` may be `null`. I think we can just add an `else` clause above, since the length of the clauses have greatly decreased thanks to your refactoring so it's not too difficult to see what's happening. Same for the other error type.

Reply:
Oh yes, good catch. I missed it. I refactored it another way to avoid adding an `else`, since we can use `isError` to determine whether the `rangeStart.isAfter(rangeEnd)` was tripped.

Reply:
I decided to go back to something similar to the pre-refactoring logic because nothing else seems to make sense after adding in the parse validation.